CMakeLists.txt

```diff
@@ -1,4 +1,10 @@
 
+# Disable in-source builds to prevent source tree corruption.
+if( "${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}" )
+  message( FATAL_ERROR "FATAL: In-source builds are not allowed.
```

Comment:
I would do this as a single-line string and let CMake do the wrapping. This prints weirdly on my terminal.
Maybe redundant to say "FATAL"?

Reply:
On my terminal, **FATAL** appears only once. Thus in my case, i don't see any redundancy:
```
CMake Error at CMakeLists.txt:4 (message):
  FATAL: In-source builds are not allowed.  You should create a separate
  directory for build files.


-- Configuring incomplete, errors occurred!
```